### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by kal2944**
### **IN THIS PART OF THE PROJECT WE HAVE MADE THE FOLLOWING CHANGES WITH @AbdimulhinYussuf3675 **

Checked  YAGNI, KISS, and DRY rules and Ruby style guide are respected.
Followed SOLID principles.
Implement the "Single-responsibility Principle".
Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
Extract some pieces of logic into a separate class(es)

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @kal2944, @AbdimulhinYussuf3675, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that should be created using peer programming therefore all the students in the group should contribute fairly to this PR, also make sure to remove the `Live demo` link from your README file

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: main.rb [hunk 1]
```diff
@@ -1,49 +1,50 @@
...
-  else
-    puts 'That is not a valid option'
+  def execute_options(option)
+    case option
+    when 1
+      @app.list_books
+    when 2
+      @app.list_people
+    when 3
+      @app.new_person
```
Comment (tresorsawasawa) [inline main.rb:30]: - [ ] As required for the project, kindly ensure that your library console app still behaves in the same as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ). Right now when I want to create a student, I get the below message:
Screenshot from 2023-04-17 17-22-49

File: main.rb [hunk 2]
```diff
@@ -1,49 +1,50 @@
...
+    when 4
+      @app.new_book
+    when 5
+      @app.new_rental
```
Comment (tresorsawasawa) [inline main.rb:34]: - [ ] As required for the project, kindly ensure that your library console app still behaves in the same as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ). Right now when I want to create a `Rental`, I get the below message:
Screenshot from 2023-04-17 17-28-56
**I suggest you check all the options of your application and test them before you request a new review.**

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @kal2944 and @AbdimulhinYussuf3675 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 
- Your README is professional :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi Team,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**